### PR TITLE
Rancher23 istio metric graph

### PR DESCRIPTION
**Pull request by fyery-chen**
This PR support Istio metric graph.

1. Get Istio metric graph through API
2. Create Istio metric expression after enabling Istio
3. Some integration test about it

TODO: After Charts PR is merge, we need to the istio template id to Rancher system charts repo.
Related PR:
https://github.com/rancher/system-charts/pull/44

### CODE REVIEW

File: pkg/api/customization/monitor/project_graph_action.go [hunk 1]
```diff
@@ -59,41 +60,65 @@ func (h *ProjectGraphHandler) QuerySeriesAction(actionName string, action *types
...
-		_, projectName := ref.Parse(graph.ProjectID)
-		refName := getRefferenceGraphName(projectName, graph.Name)
-		monitorMetrics, err := graph2Metrics(userContext, mgmtClient, clusterName, g.ResourceType, refName, graph.MetricsSelector, graph.DetailsMetricsSelector, inputParser.Input.MetricParams, inputParser.Input.IsDetails)
+		mgmtClient := h.clustermanager.ScaledContext.Management
+		istioGraphs, err := mgmtClient.ClusterMonitorGraphs(clusterName).List(metav1.ListOptions{LabelSelector: "component=istio"})
+		if err != nil {
+			return fmt.Errorf("list istio graph failed, %v", err)
+		}
+		for _, graph := range istioGraphs.Items {
+			_, projectName := ref.Parse(inputParser.Input.Filters["projectId"])
```
Comment (orangedeng) [inline pkg/api/customization/monitor/project_graph_action.go]: Can we use `inputParser.ProjectID`?

Comment (fyery-chen) [reply pkg/api/customization/monitor/project_graph_action.go]: Of couse, we can use `inputParser.ProjectID`, maybe it is better to use it. I will update it.

Comment (fyery-chen) [reply pkg/api/customization/monitor/project_graph_action.go]: Updated!

**Comment by orangedeng** [comment]
LGTM

**Comment by loganhz** [comment]
Link to https://github.com/rancher/rancher/issues/19087

File: pkg/controllers/user/monitoring/metricExpression.go [hunk 2]
```diff
@@ -5059,7 +5059,7 @@ spec:
     container_name!=""}[5m])) by (pod_name)
   legendFormat: CPU cfs throttled([[pod_name]])
   description: workload cpu cfs throttled seconds sum rate
----
+--- 
 `
```
Comment (aiwantaozi) [inline pkg/controllers/user/monitoring/metricExpression.go]: what this for?

**Comment by aiwantaozi** [comment]
Hi @fyery-chen Need to update vendor first because your first commit depends on const in the vendor

**Comment by fyery-chen** [comment]
@aiwantaozi Updated!

**Comment by aiwantaozi** [comment]
LGTM

**Comment by orangedeng** [comment]
Closing this PR in favor of PR https://github.com/rancher/rancher/pull/20486.

**Pull request closed without merging**